sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -297,6 +297,22 @@ case class GpuRangeExec(range: org.apache.spark.sql.catalyst.plans.logical.Range
     throw new IllegalStateException(s"Row-based execution should not occur for $this")
 }
 
+object unionMetrics {
```

Comment:
It is really odd to me that we are putting this here.  Could we just inline it?  It is only called in a single location, that itself is a part of a lambda.